File: Data Structures & Algorithms/sort-an-array/submission-0.java
class Solution {
    public int[] sortArray(int[] nums) {
        int[] count = new int[100001];
        for (int num : nums) {
            count[num + 50000] ++;
        }
        int i = 0;
        for (int j = 0; j < count.length; j ++) {
            int c = count[j];
            while (c -- > 0) {
                nums[i ++] = j - 50000;
            }
        }
        return nums;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        total = 1
        hasZero = False
        for num in nums:
            if num == 0:
                if hasZero:
                    return [0]*(len(nums))
                hasZero = True
                continue
            total *= num 
        res = []
        for num in nums:
            if hasZero:
                if num == 0:
                    res.append(total)
                else:
                    res.append(0)
            else:
                res.append(total // num)
        return res
        